src/hotspot/cpu/x86/x86_64.ad

```diff
@@ -6603,6 +6873,18 @@ instruct addI_rReg(rRegI dst, rRegI src, rFlagsReg cr)
   ins_pipe(ialu_reg_reg);
 %}
 
+instruct addI_rReg_ndd(rRegI dst, rRegI src1, rRegI src2)
+%{
+  predicate(UseAPX);
+  match(Set dst (AddI src1 src2));
+
```

Comment:
What if $dst and $src1 is the same registers, do you still intend to pay the penalty for 4 bytes extended EVEX prefix to save flag update?

Reply:
Currently working on enabling the various APX NDD instructions. Optimization for special cases will come next. Also, optimizations will be pursued using a data driven approach.

Reply:
Matching precedes allocation, but JIT instruction emission needs to know the register encodings. So we must demote NDD instructions to legacy encodings in applicable scenarios, you can either create macro assembly routines for demoting NDD instructions to legacy instructions, on the lines of EVEX to VEX demotions which are part of assembler or do this in peephole as @merykitty proposed.

Reply:
The various suggestions and improvements will be taken into consideration and will be incorporated in future iterations.

Reply:
Hey @vamsi-parasa , please don't bother respond on draft PR, idea here is to educate you so that you can accommodate reviewer's suggestion in next version.